cmake/Modules/FindPythonExtra.cmake

```diff
@@ -225,3 +229,7 @@ find_package_handle_standard_args(PythonExtra
   FOUND_VAR PythonExtra_FOUND
   REQUIRED_VARS ${_required_vars}
 )
+
+if(DEFINED CMAKE_SYSROOT)
+  set(PYTHON_MODULE_EXTENSION ${PythonExtra_EXTENSION_SUFFIX}${PythonExtra_EXTENSION_EXTENSION})
+endif()
```

Comment:
If we are cross-compiling, we need to bypass the PYTHON_MODULE_EXTENSION definition done by pybind11 too, or this fix will **not** have effect on libraries using pybind11 directly (aka rclpy).